rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java

```diff
@@ -536,6 +551,115 @@ void saveOldFederation_after_RSKIP123_should_save_null() {
+        when(activationsWithRskip284ActiveAnd293Inactive.isActive(ConsensusRule.RSKIP293)).thenReturn(false);
+
+        ActivationConfig.ForBlock activationsWithRskip284And293Active = mock(ActivationConfig.ForBlock.class);
+        when(activationsWithRskip284And293Active.isActive(ConsensusRule.RSKIP284)).thenReturn(true);
+        when(activationsWithRskip284And293Active.isActive(ConsensusRule.RSKIP293)).thenReturn(true);
+
+        return Stream.of(
+            Arguments.of(activationsWithRskip284And293Inactive, NetworkParameters.ID_TESTNET),
+            Arguments.of(activationsWithRskip284And293Inactive, NetworkParameters.ID_MAINNET),
+            Arguments.of(activationsWithRskip284ActiveAnd293Inactive, NetworkParameters.ID_TESTNET),
+            Arguments.of(activationsWithRskip284ActiveAnd293Inactive, NetworkParameters.ID_MAINNET),
+            Arguments.of(activationsWithRskip284ActiveAnd293Inactive, NetworkParameters.ID_TESTNET),
+            Arguments.of(activationsWithRskip284ActiveAnd293Inactive, NetworkParameters.ID_MAINNET),
+            Arguments.of(activationsWithRskip284And293Active, NetworkParameters.ID_TESTNET),
+            Arguments.of(activationsWithRskip284And293Active, NetworkParameters.ID_MAINNET)
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("provideFederationBtcUTXOsTestArguments")
+    void testGetNewFederationBtcUTXOs(ActivationConfig.ForBlock activations, String networkId) {
+
+        NetworkParameters networkParameters = NetworkParameters.fromID(networkId);
+        StorageAccessor storageAccessor = new InMemoryStorage();
+        Address btcAddress = BitcoinTestUtils.createP2PKHAddress(this.networkParameters, "test");
+
+        List<UTXO> federationUtxos = BitcoinTestUtils.createUTXOs(1, btcAddress);
+        storageAccessor.saveToRepository(NEW_FEDERATION_BTC_UTXOS_KEY.getKey(), BridgeSerializationUtils.serializeUTXOList(federationUtxos));
+
+        List<UTXO> federationUtxosAfterRskip284Activation = BitcoinTestUtils.createUTXOs(2, btcAddress);
+        storageAccessor.saveToRepository(NEW_FEDERATION_BTC_UTXOS_KEY_FOR_TESTNET_PRE_HOP.getKey(), BridgeSerializationUtils.serializeUTXOList(federationUtxosAfterRskip284Activation));
+
+        List<UTXO> federationUtxosAfterRskip293Activation = BitcoinTestUtils.createUTXOs(2, btcAddress);
+        storageAccessor.saveToRepository(NEW_FEDERATION_BTC_UTXOS_KEY_FOR_TESTNET_POST_HOP.getKey(), BridgeSerializationUtils.serializeUTXOList(federationUtxosAfterRskip293Activation));
+
+        FederationStorageProvider federationStorageProvider = createFederationStorageProvider(storageAccessor);
+
+        List<UTXO> obtainedUtxos = federationStorageProvider.getNewFederationBtcUTXOs(networkParameters, activations);
+
+        if (!networkId.equals(NetworkParameters.ID_TESTNET)) {
+            Assertions.assertEquals(federationUtxos, obtainedUtxos);
+            return;
+        }
+
+        // testnet
+        // rskip284 & rskip293 are not active
```

Comment:
According to this, RSKIP284 is activated first and RSKIP293 later. So I wouldn't pass the argument where RSKIP293 is active and RSKIP284 not

Reply:
Updated.